Zend/zend_ast.c

```diff
@@ -789,7 +789,12 @@ ZEND_API zend_result ZEND_FASTCALL zend_ast_evaluate(zval *result, zend_ast *ast
 		{
 			zend_string *class_name = zend_ast_get_str(ast->child[0]);
 			zend_string *const_name = zend_ast_get_str(ast->child[1]);
+
+			EG(exception_filename) = scope->info.user.filename;
```

Comment:
I wonder if it'd make sense to save the old values and restore them instead. 

E.g. for `public const BAR = Other::CLASS_CONSTANT + NonExistent::CLASS_CONSTANT;`, I believe the Other::CLASS_CONSTANT would trigger zend_ast_evaluate recursively, after which exception_filename would be NULL.

Reply:
Adjusted, thanks for the comments!